### PR TITLE
Flatten Pages/Components, add Constants folder, remove Magic Numbers and unncessary .js in imports

### PR DESCRIPTION
**Pull request by JrmCkh**
This PR consists of:

- Flattening frontend's Pages and Components folder by extracting out CSS files and place it under Global CSS folder
  - Use barrel pattern to better organize exports under Pages/Components sub-folder
- Improving Code Quality by adding constants folder for each microservice
- Removed extra QuestionRoute.js file under ./Question (already have QuestionRoutes.js)
- Removed magic numbers
- Removed unncessary ".js" in imports

### CODE REVIEW

**Review by anthonynmh — APPROVED** [review]
LGTM

File: User/src/UserService.js [hunk 1]
```diff
@@ -62,16 +71,18 @@ const createUser = async (email, password, confirmPassword) => {
     });
 
     // Check if the password is at least 8 characters long
-    if (password.length < 8) {
+    if (password.length < MIN_PASSWORD_LENGTH) {
       throw Object.assign(
         new Error('Password must be at least 8 characters long'),
```
Comment (venuslimm) [inline User/src/UserService.js:76]: Should we replace 8 with MIN_PASSWORD_LENGTH

File: User/src/UserService.js [hunk 2]
```diff
@@ -188,35 +207,37 @@ const changeUserPassword = async (
...
-        status: 400,
+        status: Status.BAD_REQUEST,
       });
     }
 
     // Check if the password is at least 8 characters long
-    if (newPassword.length < 8 || confirmPassword.length < 8) {
+    if (newPassword.length < MIN_PASSWORD_LENGTH || confirmPassword.length < MIN_PASSWORD_LENGTH) {
       throw Object.assign(
         new Error('Password must be at least 8 characters long'),
```
Comment (venuslimm) [inline User/src/UserService.js:224]: Should we replace 8 with MIN_PASSWORD_LENGTH

**Review by venuslimm — CHANGES_REQUESTED** [review]

File: User/src/UserService.js (hunk 1 above)
Comment (JrmCkh) [reply User/src/UserService.js:76]: 8 was replaced with MIN_PASSWORD_LENGTH, a constant, to avoid the use of magic numbers

**Review by lhy-hoyin — CHANGES_REQUESTED** [review]

File: frontend/src/components/Collaboration/index.js
[the remark this answers is not in the document]
Comment (JrmCkh) [reply frontend/src/components/Collaboration/index.js:1]: resolved

File: frontend/src/components/User/index.js
[the remark this answers is not in the document]
Comment (JrmCkh) [reply frontend/src/components/User/index.js:1]: resolved.

File: frontend/src/index.css
[the remark this answers is not in the document]
Comment (JrmCkh) [reply frontend/src/index.css:1]: resolved

File: Collaboration/src/CollaborationServer.js
[the remark this answers is not in the document]
Comment (JrmCkh) [reply Collaboration/src/CollaborationServer.js:1]: Resolved

**Comment by lhy-hoyin** [comment]
LGMT

**Review by lhy-hoyin — APPROVED** [review]
LGTM

**Review by venuslimm — APPROVED** [review]
LGTM

**Pull request merged**